### PR TITLE
Integration specs

### PR DESCRIPTION
**Pull request by tarikbouari**
In this project I and @laiifuu we implement the following features

# PR Description

- Use Capybara to write integration tests for each view in your project.
-  Test to see the username of all other users.
- Test to see the profile picture for each user.
- Test to see the number of posts each user has written.
- When I click on a user, I am redirected to that user's show page.
-Test to see the user's profile picture.
- Test to see the user's username.
- est to see the number of posts the user has written.
- Test to see the user's bio.
- Test to see the user's first 3 posts 
- Test to see a button that lets me view all of a user's posts.
- User post index page:
- Test User show index page
- Test Post index page
- Test Post show page
- Here are the screenshots of the app's logs before and after the fix of the N+1 problem 

# Before 
usersPosts1

# After 

usersPosts2

Refactor code and fix linters error

### CODE REVIEW

File: spec/views/posts/index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  it 'shows number of likes a post has' do
+    expect(page.body).to include('Likes: 0')
+  end
+
+  it 'shows the title of the post' do
+    expect(page).to have_content(@second_post.title)
+  end
+
+  it 'shows part of a post body (This is ObiWans\' message: Don\'t let the Empire win)' do
+    expect(page).to have_content(@second_post.text)
```
Comment (omar-muhamad) [inline spec/views/posts/index_spec.rb]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20first%20comments%20on%20a%20post.), Please add tests that assert you can see the first comments on a post.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.), Please add Tests that asserts you can see a section for pagination if there are more posts than fit on the view.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @tarikbouari and @laiifuu,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mukhammadrizooff — APPROVED** [review]
Hi @tarikbouari, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter errors ✔️ 
- Good implementations ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by FrederickMih — COMMENTED** [review]
## Hi :raising_hand:  @tarikbouari, 


While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request :red_circle: 

You have submitted a collaborative project, that has already been approved and merged. Since this is a collaborative project and you're repeating the module, kindly make sure to work together in a team in order to grasp the concept better. If in case you don't have a partner, kindly reach out to the student success.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Pull request merged**